lms/djangoapps/email_marketing/signals.py

```diff
@@ -273,31 +273,46 @@ def force_unsubscribe_all(sender, **kwargs):  # pylint: disable=unused-argument
     of success or failure.
 
     Args:
-        user(User): Django model of type returned from get_user_model()
+        email: Email address to unsubscribe
+        new_email (optional): Email address to change 3rd party services to for this user (used in retirement to clear
+                              personal information from the service)
     Returns:
         None
     """
-    user = kwargs.get('user', None)
+    email = kwargs.get('email', None)
+    new_email = kwargs.get('new_email', None)
 
-    if not user:
-        raise TypeError('Expected a User type, but received None.')
+    if not email:
+        raise TypeError('Expected an email address to unsubscribe, but received None.')
 
     email_config = EmailMarketingConfiguration.current()
     if not email_config.enabled:
         return
 
-    sailthru_parms = {"id": user.email, "keys": {"optout_email": "all"}}
+    sailthru_parms = {
+        "id": email,
+        "fields": {
```

Comment:
I was wrong in my initial suggestion.
I think you got this pretty close. But in order to set the value of `optout_email`, you have to be part of top level dictionary item.
So the sample full dictionary should look like this:
```
{
    "id": email,
    "optout_email": "all",
    "keys": { "email": new_email},
    "keysconflict": "merge",
    "fields": {"keys": 1, "optout_email": 1}
}
```